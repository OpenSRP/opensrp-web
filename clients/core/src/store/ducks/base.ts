import { get, keyBy, values, Dictionary } from 'lodash';
import { AnyAction, Store } from 'redux';
import SeamlessImmutable from 'seamless-immutable';

/** describes primary required properties for a client object from opensrp
 * extend this to create generic types for clients.
 */
export interface BaseClient {
    type: 'Client';
    baseEntityId: string;
}

// actions

/** CLIENTS_FETCHED action type */
export const CLIENTS_FETCHED = 'opensrp/reducer/clients/CLIENTS_FETCHED';
/** REMOVE_CLIENTS action type */
export const REMOVE_CLIENTS = 'opensrp/reducer/clients/REMOVE_CLIENTS';

/** interface for authorize action
 * ClientType - generic type - client type being handled by this function
 */
export interface FetchClientsAction<ClientType> extends AnyAction {
    clientsById: Dictionary<ClientType>;
    type: typeof CLIENTS_FETCHED;
}

/** Interface for removeClientsAction */
interface RemoveClientsAction extends AnyAction {
    clientsById: {};
    type: typeof REMOVE_CLIENTS;
}

/** Create type for clients reducer actions */
export type ClientsActionTypes<ClientType> = FetchClientsAction<ClientType> | RemoveClientsAction | AnyAction;

// factory methods for action creators

/** creates the action creator
 * ClientType - generic type - client type being handled by this function
 */
export function fetchClientsFactory<ClientType extends BaseClient>() {
    /** Fetch clients action creator
     * @param {Client []} clientsList - clients array to add to store
     * @return {FetchClientsAction} - an action to add clients to redux store
     */
    return (clientsList: ClientType[] = []): FetchClientsAction<ClientType> => ({
        clientsById: keyBy<ClientType>(clientsList, (client: ClientType) => client.baseEntityId),
        type: CLIENTS_FETCHED,
    });
}

// actions

/** removeClientsAction action */
export const removeClientsAction: RemoveClientsAction = {
    clientsById: {},
    type: REMOVE_CLIENTS,
};

// The reducer

/** interface for clients state in redux store
 * ClientType - generic type - client type being handled by this function
 */
interface ClientState<ClientType> {
    clientsById: { [key: string]: ClientType };
}

/** Create an immutable clients state
 * ClientType - generic type - client type being handled by this function
 */
export type ImmutableClientsState<ClientType> = ClientState<ClientType> &
    SeamlessImmutable.ImmutableObject<ClientState<ClientType>>;

/** initial clients-state state */
const initialState = SeamlessImmutable({
    clientsById: {},
});

/** factory function to create reducer
 * ClientType - generic type - client type being handled by this function
 */
export const reducerFactory = <ClientType>() =>
    /** the clients reducer function */
    function reducer(
        state: ImmutableClientsState<ClientType> = initialState,
        action: ClientsActionTypes<ClientType>,
    ): ImmutableClientsState<ClientType> {
        switch (action.type) {
            case CLIENTS_FETCHED:
                return SeamlessImmutable({
                    ...state,
                    clientsById: { ...state.clientsById, ...action.clientsById },
                });
            case REMOVE_CLIENTS:
                return SeamlessImmutable({
                    ...state,
                    clientsById: action.clientsById,
                });
            default:
                return state;
        }
    };

// Selectors

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getClientsByIdFactory = <ClientType>(
    reducerName: string,
): ((state: Partial<Store>) => Dictionary<ClientType>) => {
    /** returns all clients in the store as values whose keys are their respective ids
     * @param {Partial<Store>} state - the redux store
     */
    return function(state: Partial<Store>): Dictionary<ClientType> {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        return (state as any)[reducerName].clientsById;
    };
};

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getClientsArrayFactory = <ClientType>(reducerName: string) =>
    /** gets clients as an array of clients objects
     * @param {Partial<Store>} state - the redux store
     * @return {Client[]} - an array of clients objs
     */
    function(state: Partial<Store>): ClientType[] {
        const getClientsById = getClientsByIdFactory<ClientType>(reducerName);
        return values<ClientType>(getClientsById(state));
    };

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getClientByIdFactory = <ClientType>(reducerName: string) =>
    /** get a specific client by their id
     * @param {Partial<Store>} state - the redux store
     * @return {Client | null} a client obj if the id is found else null
     */
    function(state: Partial<Store>, id: string): ClientType | null {
        return get(getClientsByIdFactory<ClientType>(reducerName)(state), id) || null;
    };
