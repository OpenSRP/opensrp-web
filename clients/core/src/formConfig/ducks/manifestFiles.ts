import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
} from '../../store/ducks/baseDux';

/** manifest files interface */
export interface ManifestFilesTypes {
    created_at: string;
    id: string;
    identifier: string;
    is_draft: boolean;
    is_json_validator: boolean;
    jursdiction: string;
    label: string;
    module: string;
    version: string;
}

/** reducer name */
export const reducerName = 'manifestFiles';

/** manifest files Reducer */
const reducer = reducerFactory<ManifestFilesTypes>(reducerName);

// action
/** fetch manifest files to store action */
export const fetchManifestFiles = fetchActionCreatorFactory<ManifestFilesTypes>(reducerName, 'id');

/** clear manifest files data in store action*/
export const removeManifestFiles = removeActionCreatorFactory(reducerName);

// selectors
export const getAllManifestFilesById = getItemsByIdFactory<ManifestFilesTypes>(reducerName);
export const getManifestFilesById = getItemByIdFactory<ManifestFilesTypes>(reducerName);
export const getAllManifestFilesArray = getItemsArrayFactory<ManifestFilesTypes>(reducerName);

export default reducer;
