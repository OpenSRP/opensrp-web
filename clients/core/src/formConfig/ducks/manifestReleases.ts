import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
    getTotalRecordsFactory,
} from '../../store/ducks/baseDux';

/** json field interface */
export interface ManifestJsonFieldType {
    forms_version: string;
    identifiers: string[];
}

/** manifest releses interface */
export interface ManifestReleasesTypes {
    appId: string;
    appVersion: string;
    createdAt: string;
    identifier: string;
    json: string;
    updatedAt: string;
}

/** reducer name */
export const reducerName = 'manifestReleases';

/** manifest releases Reducer */
const reducer = reducerFactory<ManifestReleasesTypes>(reducerName);

// action
/** fetch manifest releases to store action */
export const fetchManifestReleases = fetchActionCreatorFactory<ManifestReleasesTypes>(reducerName, 'identifier');

/** clear manifest releases data in store action*/
export const removeManifestReleases = removeActionCreatorFactory(reducerName);

// selectors
export const getAllManifestReleasesById = getItemsByIdFactory<ManifestReleasesTypes>(reducerName);
export const getManifestReleasesById = getItemByIdFactory<ManifestReleasesTypes>(reducerName);
export const getAllManifestReleasesArray = getItemsArrayFactory<ManifestReleasesTypes>(reducerName);
export const getTotalManifestReleasesRecords = getTotalRecordsFactory(reducerName);

export default reducer;
