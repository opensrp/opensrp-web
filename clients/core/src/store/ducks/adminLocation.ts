import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    setTotalRecordsFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
    getTotalRecordsFactory,
} from './baseDux';

/** The reducer name */
export const reducerName = 'opensrp-web/admin/location';

/** Interface for Location object */

export interface LocationTag {
    id: number;
    active: boolean;
    name: string;
    description: string;
}
export interface Location {
    type: string;
    id: number | string;
    properties: {
        uid: string;
        type: string;
        status: string;
        parentId: string;
        name: string;
        geographicLevel: number;
        effectiveStartDate: string;
        version: number;
        parentName: string;
        tagName: string;
    };
    serverVersion: number;
    locationTags: LocationTag[];
}

/** Client Reducer */
const reducer = reducerFactory<Location>(reducerName);

// action creators
export const fetchLocationList = fetchActionCreatorFactory<Location>(reducerName, 'id');
export const removeLocationList = removeActionCreatorFactory(reducerName);
export const setTotalRecords = setTotalRecordsFactory(reducerName);

// selectors
export const getLocationListById = getItemsByIdFactory<Location>(reducerName);
export const getLocationArray = getItemsArrayFactory<Location>(reducerName);
export const getLocationById = getItemByIdFactory<Location>(reducerName);
export const getTotalRecords = getTotalRecordsFactory(reducerName);

export default reducer;
