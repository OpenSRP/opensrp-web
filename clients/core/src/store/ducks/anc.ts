import {
    reducerFactory,
    BaseClient,
    fetchClientsFactory,
    removeClientsFactory,
    getClientsArrayFactory,
    getClientByIdFactory,
    getClientsByIdFactory,
} from './baseClient';
import { FlexObject } from '../../helpers/utils';

/** describes an ANC client object */
export interface ANCClientType extends BaseClient {
    type: 'Client';
    dateCreated: string;
    serverVersion: number;
    clientApplicationVersion: number;
    clientDatabaseVersion: number;
    baseEntityId: string;
    identifiers: { [key: string]: string | null };
    addresses: FlexObject[];
    attributes: FlexObject;
    firstName: string;
    lastName: string;
    birthdate: string;
    middleName?: string;
    birthdateApprox: boolean;
    deathdateApprox: boolean;
    gender?: string;
    relationships: {
        [key: string]: string[];
    };
    id: string;
    revision: string;
}

/** reducer name for the ANC module */
const reducerName = 'ANC';

/** ANC Reducer */
const reducer = reducerFactory<ANCClientType>(reducerName);

// action
/** actionCreator returns action to to add anc records to store */
const fetchANC = fetchClientsFactory<ANCClientType>(reducerName);

// selectors
const getAllANCById = getClientsByIdFactory<ANCClientType>(reducerName);
const getANCById = getClientByIdFactory<ANCClientType>(reducerName);
const getAllANCArray = getClientsArrayFactory<ANCClientType>(reducerName);
const removeANCAction = removeClientsFactory(reducerName);

export { reducerName, removeANCAction, reducer, getAllANCArray, getANCById, getAllANCById, fetchANC };
