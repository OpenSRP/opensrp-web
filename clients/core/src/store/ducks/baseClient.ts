/** This file contains the generic materials required for
 * other client types such as client, household, child and ANC
 * generic material includes model, constants, actions, reducer and selectors
 * by giving in the data type it provide the common operatoins.
 */
import { get, keyBy, values, Dictionary } from 'lodash';
import { AnyAction, Store } from 'redux';
import SeamlessImmutable from 'seamless-immutable';
import { FlexObject } from '../../helpers/utils';

/** describes primary required properties for a client object from opensrp
 * extend this to create generic types for clients.
 */
export interface BaseClient {
    type: 'Client';
    baseEntityId: string;
}

// actions

/** CLIENTS_FETCHED action type */
export const CLIENTS_FETCHED = 'opensrp/reducer/clients/CLIENTS_FETCHED';
/** REMOVE_CLIENTS action type */
export const REMOVE_CLIENTS = 'opensrp/reducer/clients/REMOVE_CLIENTS';
/** SET_TOTAL_RECORDS action type */
export const SET_TOTAL_RECORDS = 'opensrp/reducer/clients/SET_TOTAL_RECORDS';

/** interface for authorize action
 * ClientType - generic type - client type being handled by this function
 */
export interface FetchClientsAction<ClientType> extends AnyAction {
    clientsById: Dictionary<ClientType>;
    type: typeof CLIENTS_FETCHED;
}

/** Interface for removeClientsAction */
interface RemoveClientsAction extends AnyAction {
    clientsById: {};
    type: typeof REMOVE_CLIENTS;
}

/** Interface for setTotalRecordsAction */
interface SetTotalRecordsAction extends AnyAction {
    totalRecords: number;
    type: typeof SET_TOTAL_RECORDS;
}

/** Create type for clients reducer actions */
export type ClientsActionTypes<ClientType> = FetchClientsAction<ClientType> | RemoveClientsAction | AnyAction;

// factory methods for action creators

/** creates the action creator
 * ClientType - generic type - client type being handled by this function
 */
export function fetchClientsFactory<ClientType extends BaseClient>() {
    /** Fetch clients action creator
     * @param {Client []} clientsList - clients array to add to store
     * @return {FetchClientsAction} - an action to add clients to redux store
     */
    return (clientsList: ClientType[] = []): FetchClientsAction<ClientType> => ({
        clientsById: keyBy<ClientType>(clientsList, (client: ClientType) => client.baseEntityId),
        type: CLIENTS_FETCHED,
    });
}

// actions

/** removeClientsAction action */
export const removeClientsAction: RemoveClientsAction = {
    clientsById: {},
    type: REMOVE_CLIENTS,
};

/** setTotalRecords action */
export const setTotalRecords = (totalCount: number): SetTotalRecordsAction => ({
    totalRecords: totalCount,
    type: SET_TOTAL_RECORDS,
});

// The reducer

/** interface for clients state in redux store
 * ClientType - generic type - client type being handled by this function
 */
interface ClientState<ClientType> {
    clientsById: { [key: string]: ClientType };
}

/** Create an immutable clients state
 * ClientType - generic type - client type being handled by this function
 */
export type ImmutableClientsState<ClientType> = ClientState<ClientType> &
    SeamlessImmutable.ImmutableObject<ClientState<ClientType>>;

/** initial clients-state state */
const initialState = SeamlessImmutable({
    clientsById: {},
    totalRecords: 0,
});

/** factory function to create reducer
 * ClientType - generic type - client type being handled by this function
 */
export const reducerFactory = <ClientType>(reducerName: string) =>
    /** the clients reducer function */
    function reducer(
        state: ImmutableClientsState<ClientType> = initialState,
        action: ClientsActionTypes<ClientType>,
    ): ImmutableClientsState<ClientType> {
        switch (action.type) {
            case CLIENTS_FETCHED:
                return SeamlessImmutable({
                    ...state,
                    [reducerName]: {
                        ...(state as FlexObject)[reducerName],
                        clientsById: { ...(state as FlexObject)[reducerName].clientsById, ...action.clientsById },
                    },
                });
            case REMOVE_CLIENTS:
                return SeamlessImmutable({
                    ...state,
                    [reducerName]: {
                        ...(state as FlexObject)[reducerName],
                        clientsById: { ...action.clientsById },
                    },
                });
            case SET_TOTAL_RECORDS:
                return SeamlessImmutable({
                    ...state,
                    [reducerName]: {
                        ...(state as FlexObject)[reducerName],
                        totalRecords: action.totalRecords,
                    },
                });
            default:
                return state;
        }
    };

// Selectors

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getClientsByIdFactory = <ClientType>(
    reducerName: string,
): ((state: Partial<Store>) => Dictionary<ClientType>) => {
    /** returns all clients in the store as values whose keys are their respective ids
     * @param {Partial<Store>} state - the redux store
     */
    return function(state: Partial<Store>): Dictionary<ClientType> {
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        return (state as any)[reducerName].clientsById;
    };
};

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getClientsArrayFactory = <ClientType>(reducerName: string) =>
    /** gets clients as an array of clients objects
     * @param {Partial<Store>} state - the redux store
     * @return {Client[]} - an array of clients objs
     */
    function(state: Partial<Store>): ClientType[] {
        const getClientsById = getClientsByIdFactory<ClientType>(reducerName);
        return values<ClientType>(getClientsById(state));
    };

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getClientByIdFactory = <ClientType>(reducerName: string) =>
    /** get a specific client by their id
     * @param {Partial<Store>} state - the redux store
     * @return {Client | null} a client obj if the id is found else null
     */
    function(state: Partial<Store>, id: string): ClientType | null {
        return get(getClientsByIdFactory<ClientType>(reducerName)(state), id) || null;
    };

/** factory function that creates selector
 * ClientType - generic type - client type being handled by this function
 */
export const getTotalRecordsFactory = (reducerName: string) =>
    /** returns the count of all records present in server
     * @param {Partial<Store>} state - the redux store
     * @return { number } - total records value from the store
     */
    function(state: Partial<Store>): number {
        return (state as any)[reducerName].totalRecords;
    };
