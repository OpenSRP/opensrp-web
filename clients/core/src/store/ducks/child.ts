import { Client } from './clients';
import {
    fetchActionCreatorFactory,
    removeFactory,
    setTotalRecordsFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
    getTotalRecordsFactory,
} from './baseDux';

/** The reducer name */
export const reducerName = 'opensrp-web/client-type/child';

/** Interface for Child object same as client */
export type Child = Client;

/** Client Reducer */
const reducer = reducerFactory<Child>(reducerName);

// action creators
export const fetchChildList = fetchActionCreatorFactory<Child>(reducerName, 'baseEntityId');
export const removeChildList = removeFactory(reducerName);
export const setTotalRecords = setTotalRecordsFactory(reducerName);

// selectors
export const getChildListById = getItemsByIdFactory<Child>(reducerName);
export const getChildArray = getItemsArrayFactory<Child>(reducerName);
export const getChildById = getItemByIdFactory<Child>(reducerName);
export const getTotalRecords = getTotalRecordsFactory(reducerName);

export default reducer;
