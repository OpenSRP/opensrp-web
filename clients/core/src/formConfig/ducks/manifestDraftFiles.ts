import {
    fetchActionCreatorFactory,
    removeActionCreatorFactory,
    reducerFactory,
    getItemsByIdFactory,
    getItemsArrayFactory,
    getItemByIdFactory,
} from '@opensrp/reducer-factory';
import { ManifestFilesTypes } from './manifestFiles';

/** reducer name */
export const reducerName = 'manifestDraftFiles';

/** manifest draft files Reducer */
const reducer = reducerFactory<ManifestFilesTypes>(reducerName);

// action
/** fetch manifest draft files to store action */
export const fetchManifestDraftFiles = fetchActionCreatorFactory<ManifestFilesTypes>(reducerName, 'id');

/** clear manifest draft files data in store action*/
export const removeManifestDraftFiles = removeActionCreatorFactory(reducerName);

// selectors
export const getAllManifestDraftFilesById = getItemsByIdFactory<ManifestFilesTypes>(reducerName);
export const getManifestDraftFilesById = getItemByIdFactory<ManifestFilesTypes>(reducerName);
export const getAllManifestDraftFilesArray = getItemsArrayFactory<ManifestFilesTypes>(reducerName);

export default reducer;
