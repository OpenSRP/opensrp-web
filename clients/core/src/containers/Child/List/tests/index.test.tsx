import * as React from 'react';
import { mount } from 'enzyme';
import { ChildList, ChildListProps } from '..';
import ConnectedChildList from '..';
import * as fixtures from '../../../../store/ducks/tests/fixtures';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { fetchChildList } from '../../../../store/ducks/child';

describe('containers/child/List', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    it('renders without crashing', () => {
        const wrapper = mount(<ChildList />);
        expect(wrapper.length).toBe(1);
    });

    it('renders correctly', () => {
        const mock: jest.Mock = jest.fn();
        const opensrpServiceMock: jest.Mock = jest.fn();
        const props: ChildListProps = {
            childArray: [],
            fetchChild: mock,
            opensrpService: opensrpServiceMock,
            removeChild: mock,
            totalRecords: 0,
            setTotalRecords: mock,
        };
        const wrapper = mount(<ChildList {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });

    it('renders correctly when childArray is an empty array', () => {
        const mock: jest.Mock = jest.fn();
        const opensrpServiceMock: jest.Mock = jest.fn();
        const props: ChildListProps = {
            childArray: [],
            fetchChild: mock,
            opensrpService: opensrpServiceMock,
            removeChild: mock,
            totalRecords: 0,
            setTotalRecords: mock,
        };
        const wrapper = mount(<ChildList {...props} />);
        expect(toJson(wrapper.find('Ripple'))).toMatchSnapshot('Ripple Loader');
        wrapper.unmount();
    });

    it('works correctly with the redux store', () => {
        store.dispatch(fetchChildList(fixtures.childList));
        const mock: jest.Mock = jest.fn();
        const opensrpServiceMock: jest.Mock = jest.fn();
        const props: ChildListProps = {
            childArray: [],
            fetchChild: mock,
            opensrpService: opensrpServiceMock,
            removeChild: mock,
            totalRecords: 0,
            setTotalRecords: mock,
        };
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedChildList {...props} />
                );
            </Provider>,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
