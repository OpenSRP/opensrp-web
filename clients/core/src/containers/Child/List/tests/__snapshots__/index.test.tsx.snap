// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/child/List renders correctly 1`] = `
<ChildList
  childArray={Array []}
  fetchChild={[MockFunction]}
  opensrpService={
    [MockFunction] {
      "calls": Array [
        Array [
          "undefined",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Object {
            "list": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "clientType": "child",
                    "gender": "",
                    "pageNumber": "1",
                    "pageSize": undefined,
                    "searchText": "",
                  },
                ],
              ],
              "results": Array [
                Object {
                  "type": "return",
                  "value": Promise {},
                },
              ],
            },
          },
        },
      ],
    }
  }
  removeChild={[MockFunction]}
  setTotalRecords={[MockFunction]}
  totalRecords={0}
>
  <Ripple
    borderColor="#ff5d00"
    borderStyle="solid"
    borderWidth="4px"
    height="64px"
    minHeight="80vh"
    width="64px"
  >
    <div
      className="lds-ripple-parent"
      style={
        Object {
          "minHeight": "80vh",
        }
      }
    >
      <div
        className="lds-ripple"
        style={
          Object {
            "height": "64px",
            "width": "64px",
          }
        }
      >
        <div
          style={
            Object {
              "borderColor": "#ff5d00",
              "borderStyle": "solid",
              "borderWidth": "4px",
            }
          }
        />
        <div
          style={
            Object {
              "borderColor": "#ff5d00",
              "borderStyle": "solid",
              "borderWidth": "4px",
            }
          }
        />
      </div>
    </div>
  </Ripple>
</ChildList>
`;

exports[`containers/child/List renders correctly when childArray is an empty array: Ripple Loader 1`] = `
<Ripple
  borderColor="#ff5d00"
  borderStyle="solid"
  borderWidth="4px"
  height="64px"
  minHeight="80vh"
  width="64px"
>
  <div
    className="lds-ripple-parent"
    style={
      Object {
        "minHeight": "80vh",
      }
    }
  >
    <div
      className="lds-ripple"
      style={
        Object {
          "height": "64px",
          "width": "64px",
        }
      }
    >
      <div
        style={
          Object {
            "borderColor": "#ff5d00",
            "borderStyle": "solid",
            "borderWidth": "4px",
          }
        }
      />
      <div
        style={
          Object {
            "borderColor": "#ff5d00",
            "borderStyle": "solid",
            "borderWidth": "4px",
          }
        }
      />
    </div>
  </div>
</Ripple>
`;

exports[`containers/child/List works correctly with the redux store 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ChildList)
    childArray={Array []}
    fetchChild={[MockFunction]}
    opensrpService={
      [MockFunction] {
        "calls": Array [
          Array [
            "undefined",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Object {
              "list": [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "clientType": "child",
                      "gender": "",
                      "pageNumber": "1",
                      "pageSize": undefined,
                      "searchText": "",
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": Promise {},
                  },
                ],
              },
            },
          },
        ],
      }
    }
    removeChild={[MockFunction]}
    setTotalRecords={[MockFunction]}
    totalRecords={0}
  >
    <ChildList
      childArray={
        Array [
          Object {
            "_id": "922db682a-0f31-4935-9114-c4d33d148617",
            "_rev": "v1",
            "addresses": Array [],
            "attributes": Object {
              "dynamicProperties": Object {
                "age_year_part": 2,
                "last_contact_date": "2020-01-01T00:02:01",
              },
              "residence": "6f7ca772-b368-4c3d-bd9c-011ra698203a",
            },
            "baseEntityId": "906fc91h-4cce-412a-8327-ca2315ekoeo9",
            "birthdate": 358110000088,
            "birthdateApprox": false,
            "clientApplicationVersion": 2,
            "clientDatabaseVersion": 2,
            "dateCreated": 1751437728487,
            "deathdateApprox": false,
            "firstName": "zysan",
            "identifiers": Object {
              "opensrp_id": "3117124",
            },
            "lastName": "DADA",
            "middleName": "",
            "relationships": Object {
              "family_head": Array [
                "gjio081-c153-4bcc-85c3-0b7c3d9f2263",
              ],
              "primary_caregiver": Array [
                "jklf5181-c153-4bcc-85c3-0b7c3d9f2263",
              ],
            },
            "serverVersion": 1557937728441,
            "type": "Client",
          },
          Object {
            "_id": "922db682a-0f31-4935-9114-c4d33d148617",
            "_rev": "v1",
            "addresses": Array [],
            "attributes": Object {
              "dynamicProperties": Object {
                "age_year_part": 2,
                "last_contact_date": "2020-01-01T00:02:01",
              },
              "residence": "1f7ca772-b368-4c3d-bd9c-011ra698203a",
            },
            "baseEntityId": "406fc91h-4cce-412a-8327-ca2315ekoeo9",
            "birthdate": 458110000088,
            "birthdateApprox": true,
            "clientApplicationVersion": 21,
            "clientDatabaseVersion": 1,
            "dateCreated": 1851437728487,
            "deathdateApprox": true,
            "firstName": "zysan",
            "identifiers": Object {
              "opensrp_id": "3117124",
            },
            "lastName": "DADA",
            "middleName": "",
            "relationships": Object {
              "family_head": Array [
                "gjio081-c153-4bcc-85c3-0b7c3d9f2263",
              ],
              "primary_caregiver": Array [
                "jklf5172-c153-4bcc-85c3-0b7c3d9f2263",
              ],
            },
            "serverVersion": 1257937728441,
            "type": "Client",
          },
          Object {
            "_id": "912db682a-0f31-4935-9114-c4d33d148617",
            "_rev": "v1",
            "addresses": Array [],
            "attributes": Object {
              "dynamicProperties": Object {
                "age_year_part": 2,
                "last_contact_date": "2020-01-01T00:02:01",
              },
              "residence": "9f7ca772-b368-4d39-bd9c-011ra698203a",
            },
            "baseEntityId": "306fc91h-4cce-41qa-9327-ca2315ekoeo9",
            "birthdate": 3581101103188,
            "birthdateApprox": false,
            "clientApplicationVersion": 2,
            "clientDatabaseVersion": 2,
            "dateCreated": 1751437728487,
            "deathdateApprox": true,
            "firstName": "tanvir",
            "identifiers": Object {
              "opensrp_id": "3117124",
            },
            "lastName": "DADA",
            "middleName": "",
            "relationships": Object {
              "family_head": Array [
                "gjio081-c153-4bcc-85c3-0b7c3d9f2263",
              ],
              "primary_caregiver": Array [
                "jklf5181-c153-4bcc-85c3-0b7c3d9f2263",
              ],
            },
            "serverVersion": 1557937728441,
            "type": "Client",
          },
        ]
      }
      fetchChild={[Function]}
      opensrpService={
        [MockFunction] {
          "calls": Array [
            Array [
              "undefined",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Object {
                "list": [MockFunction] {
                  "calls": Array [
                    Array [
                      Object {
                        "clientType": "child",
                        "gender": "",
                        "pageNumber": "1",
                        "pageSize": undefined,
                        "searchText": "",
                      },
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": Promise {},
                    },
                  ],
                },
              },
            },
          ],
        }
      }
      removeChild={[Function]}
      setTotalRecords={[Function]}
      totalRecords={0}
    >
      <Ripple
        borderColor="#ff5d00"
        borderStyle="solid"
        borderWidth="4px"
        height="64px"
        minHeight="80vh"
        width="64px"
      >
        <div
          className="lds-ripple-parent"
          style={
            Object {
              "minHeight": "80vh",
            }
          }
        >
          <div
            className="lds-ripple"
            style={
              Object {
                "height": "64px",
                "width": "64px",
              }
            }
          >
            <div
              style={
                Object {
                  "borderColor": "#ff5d00",
                  "borderStyle": "solid",
                  "borderWidth": "4px",
                }
              }
            />
            <div
              style={
                Object {
                  "borderColor": "#ff5d00",
                  "borderStyle": "solid",
                  "borderWidth": "4px",
                }
              }
            />
          </div>
        </div>
      </Ripple>
    </ChildList>
  </Connect(ChildList)>
  );
</Provider>
`;
